awx/main/utils/formatters.py

```diff
@@ -139,37 +139,6 @@ def reformat_data_for_log(self, raw_data, kind=None):
             data = json.loads(data)
         data_for_log = {}
 
-        def index_by_name(alist):
-            """Takes a list of dictionaries with `name` as a key in each dict
-            and returns a dictionary indexed by those names"""
-            adict = {}
-            for item in alist:
-                subdict = copy(item)
-                if 'name' in subdict:
-                    name = subdict.get('name', None)
-                elif 'path' in subdict:
-                    name = subdict.get('path', None)
-                if name:
-                    # Logstash v2 can not accept '.' in a name
-                    name = name.replace('.', '_')
-                    adict[name] = subdict
-            return adict
-
-        def convert_to_type(t, val):
```

Comment:
It's kind of silly to redefine this function on *every* log `.format()` call; I moved it below to the only block where it matters (if the performance logger is in, which I doubt many people outside of AWX developers are using).